src/ServiceManagement/RecoveryServices/Commands.RecoveryServices/Service/InvokeAzureRecoveryServicesVaultUpgrade.cs

```diff
@@ -0,0 +1,238 @@
+
+            this.WriteResponse(Properties.Resources.VaultUpgradeInProgress);
+
+            do
+            {
+                Thread.Sleep(PSRecoveryServicesClient.TimeToSleepBeforeFetchingJobDetailsAgain);
+                response = this.RecoveryServicesClient.TrackVaultUpgrade(
+                    this.VaultName,
+                    this.Location,
+                    this.ResourceType);
+
+                elapsedSeconds = DateTime.Now.Subtract(startTime).TotalSeconds;
+                record.PercentComplete = (int)elapsedSeconds * 100 / (int)taskTimeoutInSeconds;
+                this.WriteProgress(record);
+            }
+            while (response.OperationStatus == Constants.InProgress &&
+                elapsedSeconds < taskTimeoutInSeconds &&
+                !this.StopProcessingFlag);
+
+            this.WriteResponse(
+                string.Format(
+                    Properties.Resources.TrackVaultUpgradeJobResult,
+                    response.OperationResult,
+                    response.OperationStatus));
+
+            if (response.OperationStatus == Constants.Completed)
+            {
+                if (response.OperationResult == Constants.Succeeded)
+                {
+                    this.WriteResponse(
+                        string.Format(
+                            Properties.Resources.VaultUpgradeSucceded,
+                            this.ResourceType,
+                            this.VaultName));
+                }
+                else
+                {
+                    this.WriteResponse(Properties.Resources.VaultUpgradeNotSucceded);
+                }
+            }
+            else if (this.StopProcessingFlag)
+            {
+                this.WriteResponse(Properties.Resources.VaultUpgradeTerminated);
+            }
+            else
+            {
+                this.WriteResponse(Properties.Resources.VaultUpgradeTimedOut);
+            }
+        }
+
+        /// <summary>
+        /// Validating vault upgrade prerequisites.
+        /// </summary>
+        /// <returns>CheckPrerequisites response.</returns> 
+        private CheckVaultUpgradePrerequisitesResponse ValidateVaultUpgradePrerequisites()
+        {
+            CheckVaultUpgradePrerequisitesResponse response =
+                CheckVaultUpgradePrerequisitesResponse.Succeeded;
+
+            this.WriteResponse(Properties.Resources.StartingPrerequisitesCheck);
+
+            try
+            {
+                this.RecoveryServicesClient.TestVaultUpgradePrerequistes(
+                    this.VaultName,
+                    this.Location,
+                    this.ResourceType,
+                    this.TargetResourceGroupName,
+                    this.Profile.Context.Subscription.Id.ToString());
+
+                this.WriteObject(Properties.Resources.CheckPrereqSucceded);
+            }
+            catch (Exception exception)
+            {
+                this.WriteResponse(Properties.Resources.CheckPrereqFailed);
```

Comment:
So all of this has been done to improve readability.
All of our existing auxportal customers have to use powershell to upgrade the vault and we are  pretty sure that many of them will not be even familiar to powershell's way of doing things. We want the ux to be as easy and as readable to the customer as possible.

Second point is it's just a one  time thing for all the customer: just to move the vault from auxportal to ibiza.
Writeojects are written in a way to update the customer with the current status of the upgrade.
Aggregating some or all, will either introduce inconsistency or compromise with readability.